rsync/Patches/0001-gcc14-patch.diff

```diff
@@ -0,0 +1,135 @@
+diff -ru a/syscall.c b/syscall.c
+--- a/syscall.c	Tue Jan 14 13:30:32 2025
++++ b/syscall.c	Tue Jan 14 18:03:04 2025
+@@ -775,7 +775,8 @@
+ 		if (next_fd == -1) {
+ 			goto cleanup;
+ 		}
+-		if (dirfd != AT_FDCWD) close(dirfd);
++		/* XXX SmartOS - illumos AT_FDCWD is unsigned (0xffd19553) */
++		if (dirfd != (int)AT_FDCWD) close(dirfd);
+ 		dirfd = next_fd;
+ 	}
+ 
+@@ -784,7 +785,8 @@
+ 
+ cleanup:
+ 	free(path_copy);
+-	if (dirfd != AT_FDCWD) {
++	/* XXX SmartOS - illumos AT_FDCWD is unsigned (0xffd19553) */
++	if (dirfd != (int)AT_FDCWD) {
+ 		close(dirfd);
+ 	}
+ 	return retfd;
+diff -ru a/lib/sysxattrs.c b/lib/sysxattrs.c
+--- a/lib/sysxattrs.c	Sat Jan 15 20:21:01 2022
++++ b/lib/sysxattrs.c	Tue Jan 14 20:07:14 2025
+@@ -157,16 +157,19 @@
+ 	if (fstat(attrfd, &sb) < 0)
+ 		ret = -1;
+ 	else if (sb.st_size > SSIZE_MAX) {
++		/* XXX SmartOS - the above check makes the following... */
+ 		errno = ERANGE;
+ 		ret = -1;
+ 	} else if (buflen == 0)
+ 		ret = sb.st_size;
+-	else if (sb.st_size > buflen) {
++	else if (sb.st_size > (ssize_t)buflen) {
```

Comment:
if buflen is very big, then this cast will result in negative number. Since st_size can not be negative, it is safer to cast it to unsigned instead - like you below.

Reply:
My comment (about st_size being <= SSIZE_MAX) should've explained that.